packages/dds/map/src/test/memory/directory.spec.ts

```diff
@@ -57,30 +33,49 @@ describe("SharedDirectory memory usage", () => {
         // See the comment at the top of the test suite for more details.
     });
 
-    benchmarkMemory({
-        title: "Create empty directory",
-        minSampleCount: 1000,
-        benchmarkFn: async () => {
-            createLocalDirectory("testDirectory");
-        },
-    });
+    benchmarkMemory(new class implements IMemoryTestObject {
+        title = "Create empty directory";
+        private dir: SharedDirectory = createLocalDirectory("testDirectory");
+
+        async run () {
+            this.dir = createLocalDirectory("testDirectory");
+        };
+    }());
 
     const numbersOfEntriesForTests = [1000, 10_000, 100_000];
 
     numbersOfEntriesForTests.forEach((x) => {
-        benchmarkMemory({
-            title: `Add ${x} integers to a local directory`,
-            benchmarkFn: createTestForAddingIntegerEntries(x),
-        });
+        benchmarkMemory(new class implements IMemoryTestObject {
+            title = `Add ${x} integers to a local directory`;
+            private dir: SharedDirectory = createLocalDirectory("testDirectory");
 
-        benchmarkMemory({
-            title: `Add ${x} integers to a local directory, clear it`,
-            benchmarkFn: createTestForAddingIntegerEntriesAndClearing(x),
-        });
+            async run () {
+                this.dir = createLocalDirectory("testDirectory");
```

Comment:
We can remove this line, since the directory is already getting created in `beforeIteration()`. That'll keep the test measuring specifically the adding operation, like in one of the other comments.